docs/ts/primitives/cookies.mdx

```diff
@@ -0,0 +1,206 @@
+interface AuthParams {
+  sessionId: Cookie<"sessionId">;
+}
+
+// Auth handler that uses cookies
+const auth = authHandler<AuthParams, User>(async ({ sessionId }) => {
+  return validateAndGetUser(sessionId.value);
+});
+
+// Configure the gateway with the auth handler
+export const gateway = new Gateway({
+  authHandler: auth,
+});
+```
+
+## Setting Cookies in Responses
+
+You can set cookies in your API responses by including them in your response type:
+
+```ts
+import { api, Cookie } from "encore.dev/api";
+
+// Define a response type with a cookie
+interface LoginResponse {
+  user: UserData;
+  sessionId: Cookie<"sessionId">;
+}
+
+// Create an API endpoint that sets a cookie
+export const login = api<LoginParams, LoginResponse>({
+  method: "POST",
+  path: "/login",
+  expose: true,
+}, async (params): Promise<LoginResponse> => {
+  // Authenticate user
+  const user = await authenticateUser(params.username, params.password);
+  const sessionId = generateSessionId();
+
+  // Store session
+  await storeSession(sessionId, user.id);
+
+  // Return response with cookie
+  return {
+    user,
+    sessionId: {
+      value: sessionId,
+      httpOnly: true,
+      secure: true,
+      sameSite: "Strict",
+      maxAge: 60 * 60 * 24 * 7, // 7 days
+    }
+  };
+});
+```
+
+## Using Cookies in Requests
+
+When creating API endpoints, you can access cookies sent by the client by defining them in your parameter type:
+
+```ts
+import { api } from "encore.dev/api";
+
+// Define a request type with a cookie
+interface Params {
+  language?: Cookie<"language">;
+}
+
+// Create an API endpoint that uses the cookie
+export const get = api<Params, { msg: string }>(
+  {
+    method: "GET",
+    path: "/user/profile",
+    expose: true,
+  },
+  async ({ language }) => {
+    // Access the cookie value
+    const lang = language?.value ?? "en";
+    return { msg: `your language: ${lang}` };
+  },
+);```
+
+## Typed Cookie Values
+
+Encore.ts allows you to specify the type of cookie values for improved type safety. By default, cookie values are strings, but you can define cookies with different data types. If you omit the type parameter, Encore.ts will automatically use `string` as the type.
+
+### Available Cookie Types
+
+You can use the following types for cookie values:
+
+- `string`: Text data (default when generic parameter is omitted)
+- `number`: Numeric values
+- `boolean`: True/false values
+- `Date`: Date objects
+
+### How to Specify Cookie Types
+
+When defining cookies, you can specify the type using the generic parameter:
+
+```ts
+// String cookie - explicit declaration
+type StringCookie = Cookie<string, "sessionId">;
+
+// Equivalent string cookie with omitted type parameter
+type ImplicitStringCookie = Cookie<"sessionId">;
+
+// Number cookie
+type NumberCookie = Cookie<number, "userId">;
+
+// Boolean cookie
+type BooleanCookie = Cookie<boolean, "darkMode">;
+
+// Date cookie
+type DateCookie = Cookie<Date, "lastVisit">;
```

Comment:
Do we need these type aliases? I think we could get the point across using just the example `UserPreferences` example below

Reply:
yeah, its a bit weird, removed!